phoenix-core/src/main/java/org/apache/phoenix/calcite/CalciteUtils.java

```diff
@@ -1036,4 +1048,40 @@ public Void visitCall(RexCall call) {
             return null;
         }
     }
+
+    public static Object convertLiteral(SqlLiteral literal, PhoenixRelImplementor implementor) {
+        try {
+            final SchemaPlus rootSchema = Frameworks.createRootSchema(true);
+            final FrameworkConfig config = Frameworks.newConfigBuilder()
+                    .parserConfig(SqlParser.Config.DEFAULT)
+                    .defaultSchema(rootSchema).build();
+            Planner planner = Frameworks.getPlanner(config);
+
+            SqlParserPos POS = SqlParserPos.ZERO;
+            final SqlNodeList selectList =
+                    new SqlNodeList(
+                            Collections.singletonList(literal),
+                            SqlParserPos.ZERO);
+
+
+            String sql = new SqlSelect(POS, SqlNodeList.EMPTY, selectList, null, null, null, null,
+                    SqlNodeList.EMPTY, null, null, null).toString();
+            SqlNode sqlNode = planner.parse(sql);
+            sqlNode = planner.validate(sqlNode);
+            Project proj = (Project) (planner.rel(sqlNode).rel);
```

Comment:
Probably better to add an "assert" here to ensure the rel object is of Project class before casting it. Similarly, assert that this Project has only one expression for the following statement.